### PR TITLE
Add support to protect tags with group_id or user_id

### PR DESCRIPTION
**Pull request by weirdian2k3**
This resolves https://github.com/xanzy/go-gitlab/issues/1511

### CODE REVIEW

**Comment by weirdian2k3** [comment]
@svanharmelen would you be able to review this? I'm hoping get this updated so that I can get a downstream issue resolved

File: protected_tags.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ type ProtectedTag struct {
 type TagAccessDescription struct {
 	AccessLevel            AccessLevelValue `json:"access_level"`
 	AccessLevelDescription string           `json:"access_level_description"`
+	UserID                 *int             `url:"user_id,omitempty" json:"user_id,omitempty"`
+	GroupID                *int             `url:"group_id,omitempty" json:"group_id,omitempty"`
```
Comment (svanharmelen) [inline protected_tags.go:50]: I cannot see in the docs that these fields need to be here. Can you point me to somewhere (in the code) that shows these fields exist in this struct?

Additionally in a response struct like this we don't use pointers for build-in types and URL tags.

File: protected_tags.go [hunk 2]
```diff
@@ -111,8 +113,19 @@ func (s *ProtectedTagsService) GetProtectedTag(pid interface{}, tag string, opti
...
+	Name              *string                                   `url:"name" json:"name"`
+	CreateAccessLevel *AccessLevelValue                         `url:"create_access_level,omitempty" json:"create_access_level,omitempty"`
+	AllowedToCreate   *[]ProtectRepositoryTagsPermissionOptions `url:"allowed_to_create,omitempty" json:"allowed_to_create,omitempty"`
+}
+
+// ProtectRepositoryTagsPermissionOptions represents a protected tag permission option.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#protect-repository-tags
+type ProtectRepositoryTagsPermissionOptions struct {
```
Comment (svanharmelen) [inline protected_tags.go:125]: Maybe make this just a little shorter, like `TagsPermissionOptions` or something?

File: protected_tags.go [hunk 3]
```diff
@@ -111,8 +113,19 @@ func (s *ProtectedTagsService) GetProtectedTag(pid interface{}, tag string, opti
 // GitLab API docs:
 // https://docs.gitlab.com/ee/api/protected_tags.html#protect-repository-tags
 type ProtectRepositoryTagsOptions struct {
-	Name              *string           `url:"name" json:"name"`
-	CreateAccessLevel *AccessLevelValue `url:"create_access_level,omitempty" json:"create_access_level,omitempty"`
+	Name              *string                                   `url:"name" json:"name"`
+	CreateAccessLevel *AccessLevelValue                         `url:"create_access_level,omitempty" json:"create_access_level,omitempty"`
+	AllowedToCreate   *[]ProtectRepositoryTagsPermissionOptions `url:"allowed_to_create,omitempty" json:"allowed_to_create,omitempty"`
```
Comment (svanharmelen) [inline protected_tags.go:118]:
```suggestion
	AllowedToCreate   *[]*ProtectRepositoryTagsPermissionOptions `url:"allowed_to_create,omitempty" json:"allowed_to_create,omitempty"`
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]

**Review by svanharmelen — APPROVED** [review]
I'm still not sure if these (`user_id` and `group_id`) fields exist in the `TagAccessDescription` response, but I fixed up the PR and went ahead to merge it...

**Pull request merged**